base/Base.jl

```diff
@@ -248,15 +236,51 @@ include("baseext.jl")
 
 include("c.jl")
 include("ntuple.jl")
+include("abstractset.jl")
+include("bitarray.jl")
+include("bitset.jl")
 include("abstractdict.jl")
 include("iddict.jl")
 include("idset.jl")
 include("iterators.jl")
-using .Iterators: zip, enumerate, only
+using .Iterators: zip, enumerate
 using .Iterators: Flatten, Filter, product  # for generators
-using .Iterators: Stateful    # compat (was formerly used in reinterpretarray.jl)
 include("namedtuple.jl")
 
+include("ordering.jl")
+using .Order
+
+include("compiler/compiler.jl")
+
+const _return_type = Compiler.return_type
+
+# Enable compiler
+Core.eval(Compiler, quote
+include("compiler/bootstrap.jl")
+ccall(:jl_set_typeinf_func, Cvoid, (Any,), typeinf_ext_toplevel)
+
+include("compiler/parsing.jl")
+Core._setparser!(fl_parse)
+end)
+
+include("reflection2.jl")
```

Comment:
I think it would be fine to keep this file in `base/compiler/` and only load it within `Base.Compiler`. Alternatively, we could append it to `base/compiler/utilities.jl`?

Reply:
Discuss in #56185? But regardless, `reflection2.jl` has user-facing definitions that *use* the compiler, but are not used by it, so I'm not sure I'm understanding the suggestion. The need for the split is to be able to load `reflection.jl` (which has the shared reflection code) without Core.Compiler being available.

Reply:
Sorry I mixed up `reflection.jl` and `reflection2.jl`. I agree with the split that makes `reflection.jl` usable without loading `Compiler`. As for `reflection2.jl`, renaming it to `compiler_reflection.jl` might make the purpose clearer. We can continue the discussion in a new, separate PR if needed.